OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -84,6 +84,9 @@ public class Explodes : ConditionalTrait<ExplodesInfo>, INotifyKilled, INotifyDa
 	{
 		readonly IHealth health;
 		BuildingInfo buildingInfo;
+		Armament[] armaments;
+
+		bool anyArmaments;
```

Comment:
The `armaments.Length > 0` check should have no meaningful overhead and is used at most one (and maybe never!), so can we check that directly in `ChooseWeaponForExplosion` instead of saving it here?